java/client/src/main/java/compatibility/clients/jedis/ConfigurationMapper.java

```diff
@@ -0,0 +1,110 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package compatibility.clients.jedis;
+
+import glide.api.models.configuration.GlideClientConfiguration;
+import glide.api.models.configuration.NodeAddress;
+
+/**
+ * Utility class to map Jedis configurations to Valkey GLIDE configurations. This handles the
+ * translation between the two configuration systems.
+ */
+public class ConfigurationMapper {
+
+    /**
+     * Convert Jedis client configuration to GLIDE client configuration.
+     *
+     * @param host the server host
+     * @param port the server port
+     * @param jedisConfig the Jedis configuration
+     * @return corresponding GLIDE configuration
+     */
+    public static GlideClientConfiguration mapToGlideConfig(
+            String host, int port, JedisClientConfig jedisConfig) {
+        GlideClientConfiguration.GlideClientConfigurationBuilder builder =
+                GlideClientConfiguration.builder()
+                        .address(NodeAddress.builder().host(host).port(port).build())
+                        .requestTimeout(jedisConfig.getSocketTimeoutMillis());
+
+        // SSL/TLS Configuration
+        if (jedisConfig.isSsl()) {
+            builder.useTLS(true);
+
+            // Map SSL parameters if available
+            if (jedisConfig.getSslParameters() != null || jedisConfig.getSslSocketFactory() != null) {
+                // Note: GLIDE may have different SSL configuration options
+                // This is a simplified mapping - you may need to adapt based on GLIDE's actual SSL API
+                builder.useTLS(true);
+            }
+        }
+
+        // Authentication
+        if (jedisConfig.getUser() != null && jedisConfig.getPassword() != null) {
+            // GLIDE may handle authentication differently
+            // This is a placeholder for authentication mapping
+            // You'll need to adapt this based on GLIDE's authentication API
+        } else if (jedisConfig.getPassword() != null) {
+            // Password-only authentication
+            // Adapt based on GLIDE's API
+        }
+
+        // Protocol version
+        // Note: GLIDE may not support runtime protocol configuration
+        // Protocol is typically set at the connection level
+        if (jedisConfig.getRedisProtocol() != null) {
```

Comment:
Can we check if the user is using a redis protocol that GLIDE supports? Things will silently break if not.